src/Lean/Elab/Tactic/BVDecide/Frontend/BVDecide.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving
+-/
+prelude
+import Std.Sat.AIG.CNF
+import Std.Sat.AIG.RelabelNat
+import Std.Tactic.BVDecide.Bitblast
+import Std.Tactic.BVDecide.Syntax
+import Lean.Elab.Tactic.BVDecide.Frontend.BVDecide.SatAtBVLogical
+import Lean.Elab.Tactic.BVDecide.Frontend.Normalize
+import Lean.Elab.Tactic.BVDecide.Frontend.LRAT
+
+/-!
+This module provides the implementation of the `bv_decide` frontend itself.
```

Comment:
I found this file quite difficult to follow. Not really sure why; meta code is just confusing to me. There are always hidden invariants (for example that the goal mvar is of type `False`, which is established in a different file) and the "untyped" `Expr` is everywhere. Unclear whether this can be improved, I got through the file in the end...